### ISSUE TITLE
Only add title attribute to item if item text has been truncated

### ISSUE THREAD

**Issue by carmacleod** [carbon-design-system/carbon#8073]
Context menu, and other components that have items that may get truncated if the string is too long, have been using CSS to truncate the string and add an ellipsis, which is fine, but then a title is added to all items just to make sure that any truncated strings can be hovered to see the whole string. There are a couple of problems with this:
1. Some screen readers read the text content as the item's name (the full text is available to them, so they read the whole thing), and then they read the title as the description, so the user gets to hear every item string twice. (Some screen readers don't read the description if it is identical to the name, but that is not universal.)
2. For sighted users, having tooltips pop up on every item is pretty annoying, and not normal... for a context menu, for example.
3. It doesn't even help sighted keyboard-only users see the full text of a truncated item, because title attribute cannot be opened with keyboard.

Please only add a tooltip if the text has actually been truncated.

#7768 may help with the logic for this issue.

Would be good to look at all Carbon components where long items are truncated, to see whether the code can be abstracted out and used for all such items.

In addition, consider whether you really want to use title, or whether Carbon's "Icon tooltip" (name should probably be more generic) might make for a more consistent "tooltip experience" across pages using Carbon. This would fix the keyboard-only problem (3). In addition, if you don't hook up aria-describedby, then you will fix the double-speak problem too (1) (because the item will have a name - which the screen reader can already see - but it won't have a description that just duplicates the name).

**Comment by emyarod** [comment]
I believe #7768 should address this as you mentioned. based on the storybook preview in that PR are there any remaining components that have this issue aside from context menu?

**Comment by carmacleod** [comment]
@emyarod I don't know. Haven't looked at #7768 yet. Does it use a custom tooltip?

**Comment by emyarod** [comment]
currently it only applies `title` tooltips to truncated menu items, but I was targeting #6605 and #7451 after #7768 is merged

**Comment by tay1orjones** [comment]
Duplicate of https://github.com/carbon-design-system/carbon/issues/9208
